keras_nlp/utils/text_generation.py

```diff
@@ -128,29 +132,54 @@ def token_probability_fn(inputs):
     ```
 
     """
-    if not tf.executing_eagerly():
-        raise RuntimeError(
-            "`keras_nlp.utils.greedy_search` currently requires an eager "
-            "execution context. Please call `greedy_search` outside "
-            "tf.function or run `tf.config.run_functions_eagerly(True)` to run "
-            "tf.function in eager mode."
-        )
-
     prompt = validate_prompt(prompt)
 
     input_is_1d = prompt.shape.rank == 1
     if input_is_1d:
         prompt = prompt[tf.newaxis, :]
     validate_token_probability_fn(token_probability_fn, prompt)
 
-    i = prompt.shape[1]
-    while i < max_length:
-        # If the prompt has reached our desired length, exit while loop.
-        pred = token_probability_fn(prompt)
+    length = prompt.shape.as_list()[1]
+
+    # Pad the prompt with `pad_token_id` to `max_length`.
+    prompt = tf.concat(
```

Comment:
nit: maybe split this into two lines for readability?

padding = tf.fill((tf.shape(prompt)[0], max_length - length), pad_token_id)
prompt = tf.concat((prompt, padding), axis=-1)